core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1807,7 +1830,7 @@ abstract class RDD[T: ClassTag](
   /** A description of this RDD and its recursive dependencies for debugging. */
   def toDebugString: String = {
     // Get a debug description of an rdd without its children
-    def debugSelf(rdd: RDD[_]): Seq[String] = {
+    def debugSelf(rdd: RDD[_]): Seq[String] = stateLock.synchronized {
       import Utils.bytesToString
 
       val persistence = if (storageLevel != StorageLevel.NONE) storageLevel.description else ""
```

Comment:
Hi, @squito . Shall we use `getStorageLevel` instead of accessing `storageLevel` here? Then, it seems that we don't need `stateLock.synchronized` for this `debugSelf`.

Reply:
good point, thanks